includes/class-wc-order-stats.php

```diff
@@ -0,0 +1,412 @@
+			wp_schedule_event( strtotime( date( 'Y-m-d H:30:00' ) ), 'hourly', self::CRON_EVENT );
+		}
+
+		if ( ! self::$background_process ) {
+			self::$background_process = new WC_Order_Stats_Background_Process();
+		}
+	}
+
+	/**
+	 * Get order stats information.
+	 *
+	 * @param string $start_time UTC Timestamp.
+	 * @param string $end_time UTC Timestamp.
+	 * @param array  $args Optional arguments.
+	 * @return array
+	 */
+	public static function query( $start_time, $end_time, $args = array() ) {
+		global $wpdb;
+
+		$defaults = array(
+			'interval' => 'hour', // hour, week, day, month, or year.
+			'fields' => '*',
+		);
+		$args = wp_parse_args( $args, $defaults );
+
+		$selections = array(
+			'start_time' => 'MIN(start_time) as start_time',
+			'num_orders' => 'SUM(num_orders) as num_orders',
+			'num_items_sold' => 'SUM(num_items_sold) as num_items_sold',
+			'orders_gross_total' => 'SUM(orders_gross_total) as orders_gross_total',
+			'orders_coupon_total' => 'SUM(orders_coupon_total) as orders_coupon_total',
+			'orders_refund_total' => 'SUM(orders_refund_total) as orders_refund_total',
+			'orders_tax_total' => 'SUM(orders_tax_total) as orders_tax_total',
+			'orders_shipping_total' => 'SUM(orders_shipping_total) as orders_shipping_total',
+			'orders_net_total' => 'SUM(orders_net_total) as orders_net_total',
+		);
+
+		if ( is_array( $args['fields'] ) ) {
+			$keep = array();
+			foreach ( $args['fields'] as $field ) {
+				if ( isset( $selections[ $field ] ) ) {
+					$keep[ $field ] = $selections[ $field ];
+				}
+			}
+			if ( empty( $keep ) ) {
+				return array();
+			}
+			$selections = implode( ',', $keep );
+		} else {
+			$selections = implode( ',', $selections );
+		}
+
+		$table_name = $wpdb->prefix . self::TABLE_NAME;
+		$query = $wpdb->prepare(
+			'SELECT ' . $selections . ' FROM ' . $table_name . ' WHERE start_time >= %s AND start_time < %s GROUP BY ' . strtoupper( esc_sql( $args['interval'] ) ) . '(start_time);',
+			$start_time,
+			$end_time
+		);
+
+		return $wpdb->get_results( $query, ARRAY_A );
+	}
+
+	/**
+	 * Create the table that will hold the order stats information.
+	 */
+	public function install() {
+		global $wpdb;
+
+		require_once ABSPATH . 'wp-admin/includes/upgrade.php';
+
+		$collate = '';
+		if ( $wpdb->has_cap( 'collation' ) ) {
+			$collate = $wpdb->get_charset_collate();
+		}
+
+		$table_name = $wpdb->prefix . self::TABLE_NAME;
+		$sql = "CREATE TABLE $table_name (
+			start_time int(11) DEFAULT 0 NOT NULL,
+			num_orders int(11) UNSIGNED DEFAULT 0 NOT NULL,
+			num_items_sold int(11) UNSIGNED DEFAULT 0 NOT NULL,
+			orders_gross_total double DEFAULT 0 NOT NULL,
+			orders_coupon_total double DEFAULT 0 NOT NULL,
+			orders_refund_total double DEFAULT 0 NOT NULL,
+			orders_tax_total double DEFAULT 0 NOT NULL,
+			orders_shipping_total double DEFAULT 0 NOT NULL,
+			orders_net_total double DEFAULT 0 NOT NULL,
+			PRIMARY KEY (start_time)
+		) $charset_collate;";
```

Comment:
@mikejolley I haven't had time to make all of the requested changes yet. If you could please review this updated table schema one last time on Friday that would be super helpful. Thanks!

Reply:
Biggest change is a move to integer timestamps and getting rid of some unnecessary columns.

Reply:
Looks good to me, just the timestamp vs int to look at :)